packages/jest-snapshot/src/inline_snapshots.ts

```diff
@@ -60,30 +46,65 @@ const saveSnapshotsForFile = (
   snapshots: Array<InlineSnapshot>,
   sourceFilePath: Config.Path,
   prettier: any,
-  babelTraverse: Function,
+  babelTraverse: BabelTraverse,
 ) => {
   const sourceFile = fs.readFileSync(sourceFilePath, 'utf8');
 
-  // Resolve project configuration.
-  // For older versions of Prettier, do not load configuration.
-  const config = prettier.resolveConfig
-    ? prettier.resolveConfig.sync(sourceFilePath, {
-        editorconfig: true,
-      })
-    : null;
-
-  // Detect the parser for the test file.
-  // For older versions of Prettier, fallback to a simple parser detection.
-  const inferredParser = prettier.getFileInfo
-    ? prettier.getFileInfo.sync(sourceFilePath).inferredParser
-    : (config && config.parser) || simpleDetectParser(sourceFilePath);
-
-  // Format the source code using the custom parser API.
-  const newSourceFile = prettier.format(sourceFile, {
-    ...config,
-    filepath: sourceFilePath,
-    parser: createParser(snapshots, inferredParser, babelTraverse),
-  });
+  let newSourceFile;
+  if (prettier) {
+    // Resolve project configuration.
+    // For older versions of Prettier, do not load configuration.
+    const config = prettier.resolveConfig
+      ? prettier.resolveConfig.sync(sourceFilePath, {
+          editorconfig: true,
+        })
+      : null;
+
+    // Detect the parser for the test file.
+    // For older versions of Prettier, fallback to a simple parser detection.
+    const inferredParser = prettier.getFileInfo
+      ? prettier.getFileInfo.sync(sourceFilePath).inferredParser
+      : (config && config.parser) || simpleDetectParser(sourceFilePath);
+
+    // Format the source code using the custom parser API.
+    newSourceFile = prettier.format(sourceFile, {
+      ...config,
+      filepath: sourceFilePath,
+      parser: createParser(snapshots, inferredParser, babelTraverse),
```

Comment:
Might it make more sense if we choose to go down the path of using Babel AST instead of Prettier to _always_ use Babel for the AST? I don't see the need to continue with a custom Prettier parser here.

If the user has Prettier, we can simply run the formatter afterwards to keep the auto-formatting.

Reply:
I would prefer that for sure. It'd make this much simple. @SimenB mentioned he wanted to keep in the automatic formatting in one of the earlier comments. For my use case I just want to be able to avoid prettier, but maybe long term there could be something in jest config which instructs how to transform the text by pointing to a module or command, which would be used whenever snapshots change (that way the user can choose if they want prettier, tslint, eslint, tsfmt, whatever).

Would you like me to get rid of prettier completely here? Up to you guys.

Reply:
Sorry if I wasn't clear: I wasn't suggesting getting rid of Prettier completely. Instead:
1) Always do the snapshot insertions with Babel
2) Run Prettier if the user has it installed

I actually have a PR coming shortly that's the exact opposite of this PR's title. I'm working on ensuring the inserted snapshots are formatted/indented correctly. However, I'm taking special care to try and do it in a way that is compatible with the "no prettier reliance" goal in this PR.

Reply:
This seems ideal.

Always use the least opiniated (an AST parser, probably babel) then format with prettier if it exists

Reply:
@scotthovestadt coming back to this now that https://github.com/facebook/jest/pull/8277 is in. My plan is to get rid of the insertion parser, replacing it with a standard babel traverse. I'll keep in the formatting parser if prettier is installed (looking into whether it can just remain unmodified). Let me know if I've misunderstood something about your PR and that wasn't what you intended.

NB I pushed some messed-up conflicts to this branch, so I might do a couple more pushes just so I can clearly see the full diff in github's UI, so please ignore if there are a couple of red builds on this PR.